update RN Join Docs (#1009)
100mslive/100ms-docs: docs/react-native/v2/features/join.mdx

nav: 3.1
---

## Overview

> To join and interact with others in audio or video call, the user needs to join a `room`.

Reviewer: To interact with others in audio or video call, the user needs to Join a `room`.

> To join and interact with others in audio or video call, the user needs to join a `room`.

Reviewer: ```suggestion
To interact with others in audio or video call, the user needs to Join a room.
```

> When user indicates that they want to _join_ the room, your app should have -

Reviewer: ```suggestion
To _join_ the room, your app should have -
```

1. User Name - The name which should be displayed to other peers in the room.
2. [Authentication Token](../guides/token) - The Client side Authentication Token generated by the Token Service.

You can also optionally pass these fields -

1. Settings - Such as to join room with Muted Audio or Video using the `HMSTrackSettings` object.

> 2. User Metadata - This can be used to pass any additional metadata associated with the user using `metadata` of `HMSConfig` object.

Reviewer: ```suggestion
2. User Metadata - This can be used to pass any additional metadata associated with the user using `metadata` property of `HMSConfig` object.
```

## Join a Room

We'll call the `join` method on `HMSSDK` object with a config containing above fields to join the room.

> 1. First, create an instance of `HMSSDK` using the `build` function. You'll need to add Event Listener for `HMSUpdateListenerActions` which are invoked to notify updates happening in the room like a peer joins/leaves, a track got muted/unmuted, etc.

Reviewer: ```suggestion
1. First, create an instance of `HMSSDK` using the `build` function. You'll need to add Event Listeners for `HMSUpdateListenerActions` which are invoked to notify about updates happening in the room like a peer joins/leaves, a track got muted/unmuted, etc.
```

```js
import { HMSSDK } from '@100mslive/react-native-hms';

// create HMSSDK instance using the build function
const hmsInstance = await HMSSDK.build();

// add Event Listeners to subscribe to Join Success or Failure updates
hmsInstance.addEventListener(HMSUpdateListenerActions.ON_ERROR, onError);
hmsInstance.addEventListener(HMSUpdateListenerActions.ON_JOIN, onJoin);
```

2. Next, create an object of `HMSConfig` class using the available joining configurations.

```js
let config = new HMSConfig({
                    authToken: 'eyJH5c', // client-side token generated from your token service 
                    username: 'John Appleseed',

>                 }

Reviewer: ```suggestion
                });
```

```

> 3. Now, we are primed to join the room. All you have to do is calling `join` by passing the `config` object.

Reviewer: ```suggestion
3. Now, we are ready to join the room. All we have to do is call `join` method on instance of `HMSSDK` and pass it the `config` object.
```

```js
hmsInstance.join(config);
```

That's it. You have joined a room successfully 🥳

## Join with Muted Audio / Video

Joining with Muted Audio / Video is a customization that sets the microphone and camera state before joining the room.

> By default 100ms SDK keeps the microphone and camera ON but by using this feature you can decide their state according to the use case.

Reviewer: ```suggestion
By default 100ms SDK keeps the microphone and camera ON but by using this feature you can decide their state according to your use case.
```

}
```

Let's see how this can be achieved in the following steps.

1. First, for joining a Room with Muted Audio / Video `trackSettings` property should be set while building the `HMSSDK` instance as follows -

```js
// First, create the Track Settings object

> const trackSettings = getTrackSettings();

Reviewer: Should we move `getTrackSettings` function call line after function declaration and remove "First" word from comment?

        audio: audioSettings,
    });
};
```

> 2. Now, create the `HMSSDK` object by pass the `HMSTrackSettings` object created above -

Reviewer: ```suggestion
2. Now, create an instance of `HMSSDK` by passing the `HMSTrackSettings` object created above to the build function -
```

```js
// Pass the Track Settings object to the build function 
const hmsInstance = await HMSSDK.build({
    trackSettings
});

// add Event Listeners to subscribe to Join Success or Failure updates
hmsInstance.addEventListener(HMSUpdateListenerActions.ON_ERROR, onError);
hmsInstance.addEventListener(HMSUpdateListenerActions.ON_JOIN, onJoin);
```

3. Next, create an object of `HMSConfig` class using the available joining configurations.

```js
let config = new HMSConfig({
                    authToken: 'eyJH5c', // client-side token generated from your token service 
                    username: 'John Appleseed',

>                 }

Reviewer: ```suggestion
                });
```

```

> 4. Now, we are primed to join the room. All you have to do is calling `join` by passing the `config` object.

Reviewer: ```suggestion
4. Now, we are ready to join the room. All we have to do is call `join` method on instance of `HMSSDK` and pass it the `config` object.
```

```js
hmsInstance.join(config);
```

Now, the user joins the room with the microphone and the camera turned off.

## Update Listener

> 100ms SDK provides multiple Event Listeners - `HMSUpdateListenerActions` which notify the updates happening in the room like a peer joins/leaves, a track got muted/unmutes, etc.

Reviewer: ```suggestion
100ms SDK provides multiple Event Listeners - `HMSUpdateListenerActions` which notifies about the updates happening in the room like a peer joins/leaves, a track got muted/unmutes, etc.
```
